spyder_line_profiler/spyder/plugin.py

```diff
@@ -0,0 +1,133 @@
+# -*- coding: utf-8 -*-
+# ----------------------------------------------------------------------------
+# Copyright © 2021, Spyder Line Profiler contributors
+#
+# Licensed under the terms of the MIT license
+# ----------------------------------------------------------------------------
+"""
+Spyder Line Profiler 5 Plugin.
+"""
+
+# Third-party imports
+from qtpy.QtGui import QIcon
+
+# Spyder imports
+from spyder.api.plugins import Plugins, SpyderDockablePlugin
+from spyder.api.translations import get_translation
+from spyder.api.plugin_registration.decorators import on_plugin_available
+from spyder.api.plugin_registration.decorators import on_plugin_teardown
+from qtpy.QtCore import Signal
+
+# Local imports
+from spyder.plugins.mainmenu.api import ApplicationMenus
+from spyder_line_profiler.spyder.confpage import SpyderLineProfilerConfigPage
+from spyder_line_profiler.spyder.widgets import SpyderLineProfilerWidget
+from spyder_line_profiler.spyder.widgets import SpyderLineProfilerWidgetActions
+from spyder_line_profiler.spyder.widgets import is_lineprofiler_installed
+
+_ = get_translation("spyder_line_profiler.spyder")
+
+
+
+class SpyderLineProfilerActions:
+    # Triggers
+    Run = 'run_action_menu'
+    
+
+class SpyderLineProfiler(SpyderDockablePlugin):
+    """
+    Spyder Line Profiler plugin for Spyder 5.
+    """
+
+    NAME = "spyder_line_profiler"
+    REQUIRES = [Plugins.Editor]
+    OPTIONAL = [Plugins.MainMenu]
+    TABIFY = Plugins.Help
+    WIDGET_CLASS = SpyderLineProfilerWidget
+    CONF_SECTION = NAME
+    CONF_WIDGET_CLASS = SpyderLineProfilerConfigPage
+
+    # --- Signals
+    # sig_finished = Signal()
+    """This signal is emitted to inform the profile profiling has finished."""
+
+    # --- SpyderDockablePlugin API
+    # ------------------------------------------------------------------------
+    @staticmethod
+    def get_name():
+        return _("Line Profiler")
+
+    def get_description(self):
+        return _("Line profiler display for Spyder 5")
+
+    def get_icon(self):
+        return QIcon('./data/images/spyder.line_profiler.png')
+
+    def on_initialize(self):
+        self.widget = self.get_widget()
+        # self.widget.sig_finished.connect(self.sig_finished)
```

Comment:
Probably this should be uncommented? Did you face some issue when connecting the signal from the widget to the plugin?

Reply:
I found not difference at all when disconnecting the signal. Tbh. I don't understand where the signal comes into play at all. 

I'd appreciate if you have a pointer in the documentation to read about what the signals do and in which case they are emitted by Spyder.

Reply:
So I can either 
- remove the finished signals altogether, as I don't see them being connected to anything meaningful, or
-  if they have some function within Spyder that I'm not aware of I'll put them back

Reply:
I think the idea of the signal is facilitate other plugins to be aware of the line profiler finishing execution, so kind of as an extention point/way to comunicate with other plugins. However, as you say, seems like there is no much usage for it at the moment. What do you think @ccordoba12 and @jitseniesen, should we keep it or remove it?

Reply:
Checking this @skjerns lets leave the signal. Although is not used currently, you could think of it as part of the ways the plugin has to comunicate its state to not only Spyder but other plugins so it could be useful in the future.